aws/templates/application/application_lambda.ftl

```diff
@@ -50,18 +50,17 @@
                     [#assign linkDirection = linkTarget.Direction ]
 
                     [#switch linkTargetCore.Type!""]
-                        [#case "userpool"] 
-                            [@createPolicy 
-                                mode=listMode
-                                id=formatDependentPolicyId(fnId, "link", linkName)
-                                name=fnName
-                                statements=[getPolicyStatement(
-                                                "lambda:InvokeFunction",
-                                                formatLambdaArn(fnId))]
-                                roles=formatDependentIdentityPoolAuthRoleId(
-                                        formatIdentityPoolId(linkTargetCore.Tier, linkTargetCore.Component))
-                            /]
-                            [#break]
+                        [#case USERPOOL_COMPONENT_TYPE] 
+                            [#if linkTargetResources[USERPOOL_COMPONENT_TYPE].Deployed &&
```

Comment:
if you change this to  linkTargetResources[linkTargetCore,Type].Deployed,

then the code for apigateway and userpool is identical and I thing the two case blocks can collapse to one;

[#case USERPOOL_COMPONENT_TYPE]
[#case "apigateway"]
....